### PR TITLE
LIB-391 docs(topbar): add RFC

### PR DESCRIPTION
**Pull request by oxcened**
Jira: [LIB-391](https://fiscozen.atlassian.net/browse/LIB-391)

### CODE REVIEW

File: docs/RFC/topbar/2024-05-07.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+**Related components/issues:** button, icon-button, link
+
+## Summary
+
+Use the topbar when you want to display an important message, and provides actions for users to address.
+
+## Motivation
+
+Why are we doing this? What use cases does it support? What is the expected outcome?
```
Comment (Trebh) [inline docs/RFC/topbar/2024-05-07.md:17]: Either we put the answer to this question or we remove the section, this is a prompt :laughing:

Comment (oxcened) [reply docs/RFC/topbar/2024-05-07.md:17]: Must've missed it 🤣

Comment (oxcened) [reply docs/RFC/topbar/2024-05-07.md]: fixed 👍

File: docs/RFC/topbar/2024-05-07.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+| Name        | Required | Notes                                             |
+|-------------|----------|---------------------------------------------------|
+| text        | Yes      |                                                   |
+| button      | No       | Applies if type is "button" or "hybrid"           |
+| icon-button | No       | Applies if type prop is "icon-button" or "hybrid" |
+| link        | No       | Applies if type prop is "link"                    |
+
+## Drawbacks
+
+Providing the entire action component (button/icon-button/link) as a slot can be verbose, but it enables the user to completely customize it.
```
Comment (Crish15) [inline docs/RFC/topbar/2024-05-07.md:44]: I have some doubts about the use of all these slots. How do you imagine the usage by a user? For example, from what I understand I should write the following code to display a button in the action section:

```
<FzTopBar style='button'>
   <template #text>
   very very long text
   </template>
   <template #button>
       <FzButton ... />
   </template>
</FzTopBar>
```

right?

Comment (Crish15) [reply docs/RFC/topbar/2024-05-07.md:44]: Okay then you could simplify it to a slot named “action” where you would insert the button, or link, or other components.
For hybrid mode handling you could provide an additional “action-hybrid” slot (I couldn't think of any other name ahah) where you could insert the component to be displayed in mobile device alternating with the normal action slot.
With this approach we also get rid of dependencies on other components. Obviously the component becomes less tied to the design system, so I don't know if that is really what we want

ex:
```
<FzTopBar hybrid=true>
   very very long text
   <template #action>
       <FzButton ... />
   </template>
   <template #action-mobile>
     <FzList .../>
  </template>
</FzTopBar>

```
I would use the default slot for text: the main purpose of the TopBar is to display text, so I think it is not necessary to specify a specific slot or props.

Comment (oxcened) [reply docs/RFC/topbar/2024-05-07.md:44]: Yeah that looks better. The hybrid prop could be redundant as we could just check if both the action and action-mobile slots exist.

**Review by Crish15 — APPROVED** [review]

**Review by Trebh — APPROVED** [review]

**Pull request merged**